verify.js

```diff
@@ -106,10 +112,6 @@ module.exports = function (jwtString, secretOrPublicKey, options, callback) {
       return done(new JsonWebTokenError('secret or public key must be provided'));
     }
 
-    if (!hasSignature && !options.algorithms) {
-      options.algorithms = ['none'];
-    }
```

Comment:
If `secretOrPublicKey` and `options.algorithms` are both null / undefined, line 116 will error. This was previously not possible as if `secretOrPublicKey` is undefined then `hasSignature` is guaranteed to be false (per check inline 111), which means `options.algorithms` was always set to `[none]`.

If `alg === none` then we'll have already failed, but if `alg !== none` then we'll get an `can't read toString() of undefined` error. It is correct that the token isn't valid in that case, but we should return an actual invalid token error (from passing this through to `jws.verify`).

IMO, we should also avoid changing the error message unnecessarily if we were previously going to be caught by the two checks above.

So what do you think about removing the check you've added on line 83, and putting the check here instead with the old condition. This will still prevent `none` algs being verified without being specified, as we no longer default to `none` in this case:

```js
if (!hasSignature && !options.algorithms) {
    return done(new JsonWebTokenError('specify "none" in "algorithms" to verify unsigned tokens'));
}
```

Reply:
Good catch, yeah what you've said makes sense let's move the check there.